
# Домашнее задание №1
## Ивашов Игорь
### Сбер DS 




## Задание
* Собрать информацию о всех строящихся объектах на сайте "наш.дом.рф"
* Cохранить ее в pandas dataframe
* Cохранить pandas dataframe в excel
* Cохранить pandas dataframe в pickle
* Cохранить pandas dataframe в БД

## Задание *
* Написать скрипт, который скачивает фотографии всех строящихся объектов с сайта "наш.дом.рф"


Импортируем библиотеку запросов чтобы обратиться к сайту за данными

In [2]:
import requests

Получение списока объектов

In [3]:
def get_data_from_sute_ids(x):
  #вытаскиваем по 500 объектов
  offset_ = x
  limit_ = x+499 #буду по 500 собирать
  url = f'https://xn--80az8a.xn--d1aqf.xn--p1ai/%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81%D1%8B/api/kn/object?offset={offset_}&limit={limit_}&sortField=devId.devShortCleanNm&sortType=asc&objStatus=0'
  res = requests.get(url)
  if res.ok:
    #print(res.status_code)
    objects_data = res.json()
    return objects_data['data']['list'] #возвращаем данные
  else:
    print('error')
    

In [5]:
#работает минут 30 не меньше
objId = []

for n in range(0,11100,500):
    for objects in get_data_from_sute_ids(n):
    #print(objects['objId'])
        objId.append(objects['objId'])

#print(objId)
print('количество',len(objId))



[13391, 13397, 13399, 13383, 13398, 23682, 36321, 37071, 36912, 23494, 27473, 27475, 27474, 27476, 26194, 10891, 39175, 39174, 39176, 27049, 33928, 41421, 41420, 30292, 18363, 18708, 18709, 21535, 9336, 9335, 21537, 21534, 21536, 30243, 39319, 41624, 41625, 39741, 26503, 42192, 38871, 27542, 26903, 20524, 29184, 27779, 24093, 35830, 32838, 32836, 32813, 32835, 32830, 32843, 32820, 23862, 32842, 32827, 32825, 27798, 32833, 27808, 32829, 32839, 32815, 32812, 32841, 23863, 32826, 32823, 32840, 27803, 32828, 32832, 23864, 32831, 32818, 32816, 32837, 32834, 27802, 32817, 32824, 27807, 32814, 32819, 32811, 27801, 27806, 22377, 22376, 22375, 22378, 22379, 21720, 33899, 4402, 17712, 17707, 33745, 40061, 39130, 44621, 44474, 44489, 15888, 15889, 15886, 15891, 15885, 15890, 37264, 24823, 41617, 41618, 24820, 41615, 41616, 3429, 41614, 41612, 41619, 24822, 24818, 41621, 41620, 24817, 24819, 41613, 22946, 31879, 2972, 2968, 2969, 2967, 18135, 18136, 42366, 42365, 44752, 32080, 32082, 37465, 33525,

уберем дубли из id шников

In [36]:
un_objId=[] #список уникальных id объектов
#print(objId)
for x in objId:
  if x not in un_objId:
    un_objId.append(x)
print(len(un_objId))

10095


In [37]:
obj_df = pd.DataFrame({'ObjId':un_objId})
#save to csv
obj_df.to_csv('obj_Id.csv', encoding='utf-8', index=False)
#Cохранить pandas dataframe в excel
obj_df.to_excel('obj_Id.xlsx')
#Cохранить pandas dataframe в pickle
obj_df.to_pickle("obj_Id.pkl") 


In [41]:
#Cохранить pandas dataframe в БД
from sqlalchemy import create_engine
engine = create_engine('sqlite://', echo=False)

obj_df.to_sql('OBJ_ID_T', con=engine, if_exists='replace')


In [39]:
#проверка БД запрос в табличку 
engine.execute("SELECT  * FROM OBJ_ID_T").fetchall()

[(0, 13391),
 (1, 13397),
 (2, 13399),
 (3, 13383),
 (4, 13398),
 (5, 23682),
 (6, 36321),
 (7, 37071),
 (8, 36912),
 (9, 23494),
 (10, 27473),
 (11, 27475),
 (12, 27474),
 (13, 27476),
 (14, 26194),
 (15, 10891),
 (16, 39175),
 (17, 39174),
 (18, 39176),
 (19, 27049),
 (20, 33928),
 (21, 41421),
 (22, 41420),
 (23, 30292),
 (24, 18363),
 (25, 18708),
 (26, 18709),
 (27, 21535),
 (28, 9336),
 (29, 9335),
 (30, 21537),
 (31, 21534),
 (32, 21536),
 (33, 30243),
 (34, 39319),
 (35, 41624),
 (36, 41625),
 (37, 39741),
 (38, 26503),
 (39, 42192),
 (40, 38871),
 (41, 27542),
 (42, 26903),
 (43, 20524),
 (44, 29184),
 (45, 27779),
 (46, 24093),
 (47, 35830),
 (48, 32838),
 (49, 32836),
 (50, 32813),
 (51, 32835),
 (52, 32830),
 (53, 32843),
 (54, 32820),
 (55, 23862),
 (56, 32842),
 (57, 32827),
 (58, 32825),
 (59, 27798),
 (60, 32833),
 (61, 27808),
 (62, 32829),
 (63, 32839),
 (64, 32815),
 (65, 32812),
 (66, 32841),
 (67, 23863),
 (68, 32826),
 (69, 32823),
 (70, 32840),
 (71, 27803),
 (72

# Данные по определенному объекту

в итоге у нас есть лист с id объектов сайта - 10095шт (29/01/2022) obj_df

In [7]:
import pandas as pd

In [44]:
#create new df
#obj_df = pd.read_csv('obj_Id.csv')
#obj_df = pd.DataFrame({'ObjId':un_objId})
#obj_df = read_csv('obj_Id.csv')
#print (obj_df)

In [16]:
def get_data_from_sute_obj(un_objId_):
  #вытаскивает json по id   объекта недвижимости
  url = f'https://xn--80az8a.xn--d1aqf.xn--p1ai/%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81%D1%8B/api/object/{un_objId_}'
  try:
    res = requests.get(url)
  except:
    print('ops')
  return res.json()

In [17]:
objId_=un_objId[-1] #проверка по последнему
url = f'https://xn--80az8a.xn--d1aqf.xn--p1ai/%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81%D1%8B/api/object/{objId_}'
res = requests.get(url).json()
# посмотрим как это выглядит

In [ ]:
#df = pd.json_normalize(res.json())
#df

errcode  data.id  ...  data.objGreenHouseFlg  data.objEnergyEffShortDesc
0       0    28000  ...                      0                           C

[1 rows x 91 columns]

In [19]:
import time #библиотека времени

df_all = pd.DataFrame() #очищаем

for id in un_objId: #перебираем все Id из списка ID объектов
  #print(id)
  df = pd.json_normalize(get_data_from_sute_obj(id)) #получаем карточку объекта
  df_all = pd.concat([df_all, df]) #дописываем в общий датафрэйм
  time.sleep(0.1)
print (df_all.head()) #посмотрим что получилось

  errcode  data.id  data.pdId  data.developer.devId  \
0       0    13391      28101                   306   
0       0    13397      28141                   306   
0       0    13399      28141                   306   
0       0    13383      28105                   306   
0       0    13398      28141                   306   

  data.developer.devShortCleanNm data.developer.devShortNm  \
0                     2МЕН ГРУПП                2МЕН ГРУПП   
0                     2МЕН ГРУПП                2МЕН ГРУПП   
0                     2МЕН ГРУПП                2МЕН ГРУПП   
0                     2МЕН ГРУПП                2МЕН ГРУПП   
0                     2МЕН ГРУПП                2МЕН ГРУПП   

  data.developer.devFullCleanNm  data.developer.problObjCnt  \
0                    2МЕН ГРУПП                           0   
0                    2МЕН ГРУПП                           0   
0                    2МЕН ГРУПП                           0   
0                    2МЕН ГРУПП             

In [21]:
df_all

errcode  data.id  data.pdId  data.developer.devId  \
0        0    13391      28101                   306   
0        0    13397      28141                   306   
0        0    13399      28141                   306   
0        0    13383      28105                   306   
0        0    13398      28141                   306   
..     ...      ...        ...                   ...   
0        0    28071      99284                  7344   
0        0    44447     475093                  2538   
0        0    28000      91796                  7339   
0        0    44023     532547                 10854   
0        0    37797     388962                  7061   

   data.developer.devShortCleanNm data.developer.devShortNm  \
0                      2МЕН ГРУПП                2МЕН ГРУПП   
0                      2МЕН ГРУПП                2МЕН ГРУПП   
0                      2МЕН ГРУПП                2МЕН ГРУПП   
0                      2МЕН ГРУПП                2МЕН ГРУПП   
0                      2МЕН ГРУПП                2МЕН ГРУПП   
..                            ...                       ...   
0                       ЯРКИЙ МИР                 ЯРКИЙ МИР   
0               ЯРОСЛАВЛЬЗАКАЗЧИК         ЯРОСЛАВЛЬЗАКАЗЧИК   
0                            ЯРУС                      ЯРУС   
0                             ЯСН                       ЯСН   
0               ”СЗ”АЛЬФА-ИНВЕСТ”         ”СЗ”АЛЬФА-ИНВЕСТ”   

                        data.developer.devFullCleanNm  \
0                                          2МЕН ГРУПП   
0                                          2МЕН ГРУПП   
0                                          2МЕН ГРУПП   
0                                          2МЕН ГРУПП   
0                                          2МЕН ГРУПП   
..                                                ...   
0                                           ЯРКИЙ МИР   
0     СПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК ЯРОСЛАВЛЬЗАКАЗЧИК   
0                                                ЯРУС   
0   СПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК ЯРСТРОЙНЕДВИЖИМОСТЬ   
0          СПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК АЛЬФА-ИНВЕСТ   

    data.developer.problObjCnt  data.developer.buildObjCnt  \
0                            0                           5   
0                            0                           5   
0                            0                           5   
0                            0                           5   
0                            0                           5   
..                         ...                         ...   
0                            0                           5   
0                            0                           1   
0                            0                           1   
0                            0                           1   
0                            0                           1   

    data.developer.comissObjCnt  ... data.metro.color data.metro.time  \
0                             0  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
..                          ...  ...              ...             ...   
0                             0  ...          #943E90       20.716667   
0                             3  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
0                             0  ...              NaN             NaN   
0                             1  ...              NaN             NaN   

   data.metro.isWalk data.metro.colors data.transportDistIndex  \
0                NaN               NaN                     NaN   
0                NaN               NaN                     NaN   
0                NaN               NaN                     NaN   
0 

In [24]:
#save to csv
df_all.to_csv('df_obj_all.csv',encoding='utf-8',index=False)
#Cохранить pandas dataframe в excel
df_all.to_excel('df_obj_all.xlsx')
#Cохранить pandas dataframe в pickle
df_all.to_pickle("df_obj_all.pkl") 


In [26]:
#save to DB SQLITE
from sqlalchemy import create_engine
engine = create_engine('sqlite://', echo=False)

In [27]:

df_all.to_sql('OBJ_ALL_T', con=engine, if_exists='replace')

InterfaceError: (sqlite3.InterfaceError) Error binding parameter 51 - probably unsupported type.
[SQL: INSERT INTO "OBJ_ALL_T" ("index", errcode, "data.id", "data.pdId", "data.developer.devId", "data.developer.devShortCleanNm", "data.developer.devShortNm", "data.developer.devFullCleanNm", "data.developer.problObjCnt", "data.developer.buildObjCnt", "data.developer.comissObjCnt", "data.developer.regRegionDesc", "data.developer.devPhoneNum", "data.developer.devSite", "data.developer.devEmail", "data.developer.devInn", "data.developer.devOgrn", "data.developer.devKpp", "data.developer.devLegalAddr", "data.developer.devFactAddr", "data.developer.lastRpdId", "data.developer.fundGuarantyFlg", "data.developer.devOrgRegRegionCd", "data.developer.devEmplMainFullNm", "data.developer.developerGroupName", "data.developer.orgForm.id", "data.developer.orgForm.fullForm", "data.developer.orgForm.shortForm", "data.developer.companyGroupId", "data.developer.objGuarantyEscrowFlg", "data.developer.govFundFlg", "data.region", "data.address", "data.floorMin", "data.floorMax", "data.objElemLivingCnt", "data.objReady100PercDt", "data.wallMaterialShortDesc", "data.objLkFinishTypeDesc", "data.objLkFreePlanDesc", "data.objElemParkingCnt", "data.objSquareLiving", "data.rpdNum", "data.rpdPdfLink", "data.rpdIssueDttm", "data.objLkLatitude", "data.objLkLongitude", "data.objCompensFundFlg", "data.objProblemFlg", "data.objLkClassDesc", "data.loadDttm", "data.photoRenderDTO", "data.objGuarantyEscrowFlg", "data.objectType", "data.miniUrl", "data.residentialBuildings", "data.newBuildingId", "data.objFloorCnt", "data.objFlatCnt", "data.objFlatSq", "data.objNonlivElemCnt", "data.quartography.objLivElem1KCnt", "data.quartography.objLivElem2KCnt", "data.quartography.objLivElem3KCnt", "data.quartography.objLivElem4KCnt", "data.quartography.objLivElemCnt", "data.quartography.objLivElemEntrCnt", "data.quartography.objLivElemEntrFloorAvg", "data.objStatus", "data.isAvailableWantLiveHere", "data.objTransferPlanDt", "data.objLivCeilingHeight", "data.objInfrstrBicycleLaneFlg", "data.objInfrstrPlaygrndCnt", "data.objInfrstrSportGroundCnt", "data.objInfrstrTrashAreaCnt", "data.objInfrstrObjPrkngCnt", "data.objInfrstrNotObjPrkngCnt", "data.objInfrstrRampFlg", "data.objInfrstrCurbLoweringFlg", "data.objElevatorWheelchairCnt", "data.objElevatorPassengerCnt", "data.objElevatorCargoCnt", "data.soldOutPerc", "data.objPriceAvg", "data.generalContractorNm", "data.nonlivFirstFloor", "data.objectTransportInfo", "data.conclusion", "data.objLkSaleDepEmail", "data.objGreenHouseFlg", "data.objEnergyEffShortDesc", "data.nameObj", "data.metro.id", "data.metro.name", "data.metro.line", "data.metro.color", "data.metro.time", "data.metro.isWalk", "data.metro.colors", "data.transportDistIndex", "data.complexShortNm", "data.developer.bankruptStage.bankruptStageCd", "data.developer.bankruptStage.bankruptStageDesc", "data.developer.bankruptStage.bankruptLawUrl", "data.developer.orgBankruptMsgDttm") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ((0, '0', 13391, 28101, 306, '2МЕН ГРУПП', '2МЕН ГРУПП', '2МЕН ГРУПП', 0, 5, 0, 'Тюменская область', '+7(345)279-18-88', '2mengroup.ru', 'info@2mengroup.ru', '7701651356', '1067746424899', '720301001', 'Тюменская область, город Тюмень, улица Комсомольская дом 75/5', 'Тюменская область, город Тюмень, улица Комсомольская 75/5', 551365, 1, 72.0, 'Киселев Сергей Михайлович', '2МЕН ГРУПП ДЕВЕЛОПМЕНТ', 1, 'Акционерное общество', 'АО', 699421001.0, 0, 0, 72, 'г Тюмень, ул Профсоюзная', 3, 12, 44, '2022-06-30', 'Другое', 'По желанию заказчика', 'Есть', 41.0, 4293.67, '72-000096', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/93e09cbc-021d-4f14-a940-20f7ff398171', '14-01-2022 15:59', '57.1541', '65.5595', 1, 0, 'Элитный', '29-01-2022 01:09', [{'objId': 13391, 'pdId': 28101, 'rpdId': 551365, 'devId': 306, 'ordNum': 1, 'objRenderPhotoSize': 310482, 'objRenderPhotoUrl': 'https://xn--80az8a.xn ... (4596 characters truncated) ... RenderPhotoNm': 'IMG_20210512_0944249.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/09ff8156-bcbb-4033-a36a-dd45abd0b430', 1, 37219, 12, 44, '4293.67', 26.0, 11.0, 16.0, 12.0, 5.0, 44.0, 2.0, 4.0, 0, 1, '30-08-2022 12:00', '2,8', 0, 1.0, 1.0, 1.0, 0.0, 9.0, 1, 1, 0.0, 2.0, 2.0, 0.451017427981191, 55893.0, 'ООО Транспромжилстрой-2005 (ИНН: 7204097730)', 1.0, [], 0, 'info@2mengroup.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 13397, 28141, 306, '2МЕН ГРУПП', '2МЕН ГРУПП', '2МЕН ГРУПП', 0, 5, 0, 'Тюменская область', '+7(345)279-18-88', '2mengroup.ru', 'info@2mengroup.ru', '7701651356', '1067746424899', '720301001', 'Тюменская область, город Тюмень, улица Комсомольская дом 75/5', 'Тюменская область, город Тюмень, улица Комсомольская 75/5', 551365, 1, 72.0, 'Киселев Сергей Михайлович', '2МЕН ГРУПП ДЕВЕЛОПМЕНТ', 1, 'Акционерное общество', 'АО', 699421001.0, 0, 0, 72, 'г Тюмень, ул Профсоюзная, д. 56', 6, 12, 65, '2024-12-31', 'Другое', 'Без отделки', 'Нет', 0.0, 6248.51, '72-000097', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/5b6a4f4b-8f80-4595-a53c-96f1354d69b3', '14-01-2022 16:00', '57.1519', '65.5544', 0, 0, 'Элитный', '29-01-2022 01:09', [{'objId': 13397, 'pdId': 28141, 'rpdId': 551292, 'devId': 306, 'ordNum': 1, 'objRenderPhotoSize': 1495964, 'objRenderPhotoUrl': 'https://xn--80az8a.x ... (1481 characters truncated) ... RenderPhotoNm': 'IMG_20210512_0949152.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/cb641b0a-8580-4cbd-9f0c-1dca386bdb97', 1, 37201, 12, 65, '6248.51', 70.0, 0.0, 24.0, 35.0, 6.0, 65.0, 4.0, 3.0, 0, 1, '28-02-2025 12:00', '2,96', 0, 1.0, 1.0, 1.0, 11.0, 10.0, 1, 1, 0.0, 5.0, 4.0, 0.99998399618469, 44897.0, 'ООО Транспромжилстрой-2005 (ИНН: 7204097730)', 0.0, [], 0, 'info@2mengroup.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 13399, 28141, 306, '2МЕН ГРУПП', '2МЕН ГРУПП', '2МЕН ГРУПП', 0, 5, 0, 'Тюменская область', '+7(345)279-18-88', '2mengroup.ru', 'info@2mengroup.ru', '7701651356', '1067746424899', '720301001', 'Тюменская область, город Тюмень, улица Комсомольская дом 75/5', 'Тюменская область, город Тюмень, улица Комсомольская 75/5', 551365, 1, 72.0, 'Киселев Сергей Михайлович', '2МЕН ГРУПП ДЕВЕЛОПМЕНТ', 1, 'Акционерное общество', 'АО', 699421001.0, 0, 0, 72, 'г Тюмень, ул Профсоюзная, д. 56', 1, 1, 0, '2024-12-31', 'Другое', 'Без отделки', 'Нет', 125.0, 0.0, '72-000097', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/5b6a4f4b-8f80-4595-a53c-96f1354d69b3', '14-01-2022 16:00', '57.1519', '65.5544', 0, 0, 'Элитный', '29-01-2022 01:09', [{'objId': 13399, 'pdId': 28141, 'rpdId': 551292, 'devId': 306, 'ordNum': 1, 'objRenderPhotoSize': 996519, 'objRenderPhotoUrl': 'https://xn--80az8a.xn ... (146 characters truncated) ... enderPhotoNm': 'IMG_20210512_095117.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Нежилое', 'knFlg': 0, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/6c612978-9eec-4571-a24d-90c40b806c66', 0, 34754, 1, 0, '0.0', 0.0, None, None, None, None, None, None, None, 0, 1, '28-02-2025 12:00', None, 0, 0.0, 0.0, 1.0, 0.0, 0.0, 1, 1, 0.0, 8.0, 7.0, 0.917242870424423, 50768.0, 'ООО Транспромжилстрой-2005 (ИНН: 7204097730)', 1.0, [], 0, 'info@2mengroup.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 13383, 28105, 306, '2МЕН ГРУПП', '2МЕН ГРУПП', '2МЕН ГРУПП', 0, 5, 0, 'Тюменская область', '+7(345)279-18-88', '2mengroup.ru', 'info@2mengroup.ru', '7701651356', '1067746424899', '720301001', 'Тюменская область, город Тюмень, улица Комсомольская дом 75/5', 'Тюменская область, город Тюмень, улица Комсомольская 75/5', 551365, 1, 72.0, 'Киселев Сергей Михайлович', '2МЕН ГРУПП ДЕВЕЛОПМЕНТ', 1, 'Акционерное общество', 'АО', 699421001.0, 0, 0, 72, 'г Тюмень', 4, 20, 68, '2022-06-30', 'Другое', 'По желанию заказчика', 'Есть', 29.0, 6287.45, '72-000095', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/b9c08684-a0a7-4d2c-8ae7-18a33c8a7e63', '14-01-2022 16:00', '57.153', '65.5343', 1, 0, 'Элитный', '29-01-2022 01:09', [{'objId': 13383, 'pdId': 28105, 'rpdId': 551362, 'devId': 306, 'ordNum': 1, 'objRenderPhotoSize': 1433738, 'objRenderPhotoUrl': 'https://xn--80az8a.x ... (2821 characters truncated) ... RenderPhotoNm': 'IMG_20210512_0951216.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/0dca6482-02ba-4d56-a861-7a17db04c957', 1, 39024, 20, 68, '6287.45', 10.0, 13.0, 21.0, 34.0, 0.0, 68.0, 1.0, 6.0, 0, 1, '31-08-2022 12:00', '2,8', 0, 1.0, 1.0, 1.0, 0.0, 11.0, 1, 1, 0.0, 5.0, 1.0, 0.398352273179111, 51344.0, 'ООО Транспромжилстрой-2005 (ИНН: 7204097730)', 1.0, [], 0, 'info@2mengroup.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 13398, 28141, 306, '2МЕН ГРУПП', '2МЕН ГРУПП', '2МЕН ГРУПП', 0, 5, 0, 'Тюменская область', '+7(345)279-18-88', '2mengroup.ru', 'info@2mengroup.ru', '7701651356', '1067746424899', '720301001', 'Тюменская область, город Тюмень, улица Комсомольская дом 75/5', 'Тюменская область, город Тюмень, улица Комсомольская 75/5', 551365, 1, 72.0, 'Киселев Сергей Михайлович', '2МЕН ГРУПП ДЕВЕЛОПМЕНТ', 1, 'Акционерное общество', 'АО', 699421001.0, 0, 0, 72, 'г Тюмень, ул Профсоюзная, д. 56', 7, 10, 43, '2024-12-31', 'Другое', 'Без отделки', 'Нет', 0.0, 4812.98, '72-000097', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/5b6a4f4b-8f80-4595-a53c-96f1354d69b3', '14-01-2022 16:00', '57.1519', '65.5544', 0, 0, 'Элитный', '29-01-2022 01:09', [{'objId': 13398, 'pdId': 28141, 'rpdId': 551292, 'devId': 306, 'ordNum': 1, 'objRenderPhotoSize': 1325881, 'objRenderPhotoUrl': 'https://xn--80az8a.x ... (1890 characters truncated) ... lg': 0, 'objRenderPhotoNm': 'IMG_8924.JPG', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/ddfe3c88-edc4-4b78-b1b5-68f8ea2877fc', 1, 37165, 10, 43, '4812.98', 48.0, 0.0, 11.0, 19.0, 13.0, 43.0, 3.0, 3.0, 0, 1, '28-02-2025 12:00', '2,96', 0, 1.0, 1.0, 1.0, 4.0, 15.0, 1, 1, 0.0, 3.0, 3.0, 1.0, 47886.0, 'ООО Транспромжилстрой-2005 (ИНН: 7204097730)', 0.0, [], 0, 'info@2mengroup.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 23682, 64486, 6811, '84 ВЫСОТА', '84 ВЫСОТА', '84 ВЫСОТА', 0, 1, 6, 'Санкт-Петербург город', '+7(812)309-84-09', 'www.84visota.ru', '84visota@mail.ru', '7806536079', '1147847382627', '780601001', 'город Санкт-Петербург, улица Белорусская дом 12 корпус 1 литера А помещение 1Н', 'г Санкт-Петербург ул. Белорусская д. 12 кор. 1 стр. лит. А помещение 1Н', 529403, 1, 78.0, 'Искрицкий Максим Николаевич', '84 Высота', 4, 'Общество с ограниченной ответственностью', 'ООО', 5562608001.0, 0, 0, 47, 'д Агалатово', 10, 10, 54, '2022-06-30', 'Блоки', 'Без отделки', 'Нет', 0.0, 2748.15, '47-000416', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/0432f9f8-a629-41ad-a4cc-bef3e2dcab73', '22-12-2021 13:00', '60.2152', '30.3089', 1, 0, 'Типовой', '29-01-2022 01:09', [{'objId': 23682, 'pdId': 64486, 'rpdId': 529403, 'devId': 6811, 'ordNum': 1, 'objRenderPhotoSize': 267432, 'objRenderPhotoUrl': 'https://xn--80az8a.x ... (127 characters truncated) ... HiddenFlg': 0, 'objRenderPhotoNm': '2.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 0, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/1e040c60-4f2e-45dc-9a72-e5e4fc103fe9', 1, 25764, 10, 54, '2748.15', 1.0, 27.0, 18.0, 9.0, 0.0, 54.0, 1.0, 6.0, 0, 1, '31-12-2022 12:00', '2,73', 0, 1.0, 1.0, 1.0, 168.0, 15.0, 1, 1, 0.0, 1.0, 0.0, 0.998908356530757, 54859.0, 'ООО Строительно-Монтажное Управление № 73 (ИНН: 7814244129)', 0.0, [], 0, '84visota@mail.ru', 0, 'B', 'Жилой комплекс "84 высота"', None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 36321, 319464, 10195, 'CЗ АЗИЯ-КОНТРАКТ', 'CЗ АЗИЯ-КОНТРАКТ', 'CПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК АЗИЯ-КОНТРАКТ', 0, 1, 0, 'Приморский край', '+7(914)196-21-66', 'lasto4ka.ru', '1962166@mail.ru', '2536319917', '1192536030169', '253601001', 'Приморский край, город Владивосток, улица Прапорщика Комарова дом 58', 'край Приморский, г Владивосток, ул Прапорщика Комарова д. 58', 546063, 0, 25.0, 'Жимолдинов Михаил Михайлович', None, 4, 'Общество с ограниченной ответственностью', 'ООО', None, 1, 0, 25, 'г Владивосток, район Ленинский, ул 10-я Рабочая, д. 80', 25, 25, 120, '2022-03-31', 'Монолит', 'Без отделки', 'Есть', 0.0, 8002.4, '25-000337', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/4162f2a7-a799-43d2-99c6-4c998ae91ce3', '10-01-2022 06:24', '43.1258', '131.9341', 0, 0, 'Комфорт', '29-01-2022 01:09', [{'objId': 36321, 'pdId': 319464, 'rpdId': 546063, 'devId': 10195, 'ordNum': 1, 'objRenderPhotoSize': 3286107, 'objRenderPhotoUrl': 'https://xn--80az8 ... (992 characters truncated) ... iddenFlg': 0, 'objRenderPhotoNm': '42.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 1, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/3a313534-1dfc-4c70-9e38-97740c0055ee', 1, 116538, 25, 120, '8002.4', 85.0, 120.0, 0.0, 0.0, 0.0, 120.0, 1.0, 6.0, 0, 1, '31-07-2022 12:00', '2,7', 0, 1.0, 1.0, 2.0, 0.0, 0.0, 0, 0, 0.0, 3.0, 2.0, 0.318940567829651, 129888.0, 'ООО "Тихоокеанская строительная компания" (ИНН: 2539079741)', 0.0, [], 0, 'dom@lasto4ka.ru', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 37071, 362705, 10391, 'CЗ РЕГИОНСТРОЙ', 'CЗ РЕГИОНСТРОЙ', 'CПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК РЕГИОНСТРОЙ', 0, 1, 1, 'Мордовия Республика', '+7(927)170-79-16', 'жквиктория.рф', 'ooo.regionstroy13@yandex.ru', '1326253143', '1191326001492', '132601001', 'Мордовия Республика, город Саранск, ул Пролетарская дом 130 офис 7', 'Республика Мордовия, г. Саранск, ул. Пролетарская, д. 130, офис 7', 544122, 0, 13.0, 'Сорокина Любовь Николаевна', None, 4, 'Общество с ограниченной ответственностью', 'ООО', None, 1, 0, 13, 'г Саранск, проспект Ленина, д. 29', 19, 19, 197, '2022-09-30', 'Монолит-кирпич', 'Без отделки', 'Нет', 0.0, 10520.55, '13-000155', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/32c0cf16-a0fd-4f38-908b-8e1fdfa1b03b', '10-01-2022 09:38', '54.1913', '45.1895', 0, 0, 'Комфорт', '29-01-2022 01:09', [{'objId': 37071, 'pdId': 362705, 'rpdId': 544122, 'devId': 10391, 'ordNum': 1, 'objRenderPhotoSize': 5254330, 'objRenderPhotoUrl': 'https://xn--80az8 ... (1030 characters truncated) ...  'objRenderPhotoNm': 'classic_black22.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 1, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/9c1bf997-95ab-4b0e-a4f3-945b2f4612f0', 1, 117407, 19, 197, '10520.55', 23.0, 114.0, 67.0, 16.0, 0.0, 197.0, 1.0, 12.0, 0, 1, '31-10-2022 12:00', '2,76 - 3,33', 0, 1.0, 1.0, 1.0, 76.0, 60.0, 0, 1, 0.0, 3.0, 0.0, 0.324335704882349, 49021.0, 'ООО ПродТорг (ИНН: 9725037861)', 1.0, [], 0, 'otichinina@yandex.ru', 0, 'B', 'Жилой комплекс "Виктория" ЖК "Виктория"', None, None, None, None, None, None, None, None, None, None, None, None, None)  ... displaying 10 of 10095 total bound parameter sets ...  (0, '0', 44023, 532547, 10854, 'ЯСН', 'ЯСН', 'СПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК ЯРСТРОЙНЕДВИЖИМОСТЬ', 0, 1, 0, 'Ярославская область', '+7(485)220-00-07', 'www.ysz76.ru', 'SECRETAR@TSZ76.RU', '7604299435', '1167627057620', '760401001', 'Ярославская область, город Ярославль, улица Лисицына дом 7 помещение 9', 'Ярославская область, город Ярославль, улица Лисицына, Дом 7, Помещение 9.', 543360, 0, 76.0, 'Ерин Александр Михайлович', None, 4, 'Общество с ограниченной ответственностью', 'ООО', None, 1, 0, 76, 'г Ярославль, строение 4', 10, 10, 35, '2022-03-31', 'Кирпич', 'Под чистовую', 'Нет', 0.0, 1626.45, '76-000542', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/bf34e033-a971-48bd-ad34-278ba215befb', '11-01-2022 17:17', '57.5975', '39.8837', 0, 0, 'Типовой', '29-01-2022 01:09', [{'objId': 44023, 'pdId': 532547, 'rpdId': 543360, 'devId': 10854, 'ordNum': 1, 'objRenderPhotoSize': 5830944, 'objRenderPhotoUrl': 'https://xn--80az8 ... (992 characters truncated) ... iddenFlg': 0, 'objRenderPhotoNm': '32.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 1, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/7d02e34d-4d04-45df-b1fc-152ff058e4f7', 1, 124612, 10, 35, '1626.45', 0.0, 17.0, 18.0, 0.0, 0.0, 35.0, 1.0, 4.0, 0, 1, '28-02-2022 12:00', '2,6', 0, 1.0, 1.0, 1.0, 4.0, 0.0, 1, 1, 0.0, 1.0, 0.0, None, None, 'ООО Строительная компания Гермес (ИНН: 7604129627)', 0.0, [], 0, 'SECRETAR@TSZ76.RU', 0, 'B', None, None, None, None, None, None, None, None, None, None, None, None, None, None), (0, '0', 37797, 388962, 7061, '”СЗ”АЛЬФА-ИНВЕСТ”', '”СЗ”АЛЬФА-ИНВЕСТ”', 'СПЕЦИАЛИЗИРОВАННЫЙ ЗАСТРОЙЩИК АЛЬФА-ИНВЕСТ', 0, 1, 1, 'Пермский край', '+7(342)263-01-83', 'www.good-time-prm.ru', 'alfainvest.prm@gmail.com', '5904252674', '1115904011750', '590401001', 'Пермский край, город Пермь, улица Куйбышева дом 50 корпус А офис 304', 'край Пермский город Пермь улица Уральская д. 95 оф. 8', 550092, 0, 59.0, 'Дудецкий Андрей Иванович', None, 4, 'Общество с ограниченной ответственностью', 'ООО', None, 1, 0, 59, 'г Пермь, улица Уинская, д. 2, литера Б', 2, 16, 110, '2023-03-31', 'Монолит-кирпич', 'По желанию заказчика', 'Есть', 20.0, 5524.2, '59-000606', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/b352c3f1-5190-4a91-ac9e-28b66921d8f1', '11-01-2022 14:20', '58.0078', '56.296', 0, 0, 'Комфорт', '29-01-2022 01:09', [{'objId': 37797, 'pdId': 388962, 'rpdId': 550092, 'devId': 7061, 'ordNum': 1, 'objRenderPhotoSize': 2074521, 'objRenderPhotoUrl': 'https://xn--80az8a ... (1895 characters truncated) ... g': 0, 'objRenderPhotoNm': 'page_0184.jpg', 'objReadyDesc': 'Строится', 'objBuildTypeShortDesc': 'Жилое', 'knFlg': 1, 'loadDttm': '29-01-2022 00:54'}], 1, 'normal', 'https://xn--80az8a.xn--d1aqf.xn--p1ai/api/ext/file/32922c7f-b7b9-4a47-a9e5-1e4bd7a01b37', 1, 118202, 16, 110, '5524.2', 25.0, 46.0, 44.0, 19.0, 1.0, 110.0, 1.0, 9.0, 0, 1, '30-09-2023 12:00', '2,7 - 3', 0, 1.0, 1.0, 1.0, 9.0, 0.0, 1, 1, 0.0, 1.0, 1.0, 0.899451504290214, 89274.0, 'ООО Специализированный застройщик "Империя" (ИНН: 5906162761)', 1.0, [], 1, 'alfainvest.prm@gmail.com', 0, 'B', 'Лофт на Уинской "Хорошее время"', None, None, None, None, None, None, None, None, None, None, None, None, None))]
(Background on this error at: https://sqlalche.me/e/14/rvf5)

## Написать скрипт, который скачивает фотографии всех **строящихся** объектов с сайта "наш.дом.рф"

In [28]:
import pandas as pd
import requests
def save_img_to_disk(objId_):
    #по известному адресу отправляем запрос id объекта
    url_obj = f'https://xn--80az8a.xn--d1aqf.xn--p1ai/%D1%81%D0%B5%D1%80%D0%B2%D0%B8%D1%81%D1%8B/api/object/{objId_}'
    #json сохраняем в датафрэйм
    photo_data_url = pd.json_normalize(requests.get(url_obj).json())
    url=''
    file_name =''
    #так как в одном объекте хранится несколько фотографий используем цикл для перебора всех фотографий
    for dpr in range(len(photo_data_url['data.photoRenderDTO'][0])):
        if photo_data_url['data.photoRenderDTO'][0][dpr]['objReadyDesc']=='Строится': #условие что строится объект
            url = (photo_data_url['data.photoRenderDTO'][0][dpr]['objRenderPhotoUrl']) #url до картинки
            id = str(photo_data_url['data.photoRenderDTO'][0][dpr]['objId']) #id объекта для нового названия файла
            file_end = str(photo_data_url['data.photoRenderDTO'][0][dpr]['objRenderPhotoNm'])# старое название файла
            file_name = id+"_"+file_end
            img_data = requests.get(url).content
            with open(file_name, 'wb') as handler:
                handler.write(img_data)



In [34]:
for idd in df_all['data.id'][0:10]: # проверим на первых 10 объектах
    save_img_to_disk(idd)